src/wasm-validator.h

```diff
@@ -46,13 +46,15 @@
 namespace wasm {
 
 // Print anything that can be streamed to an ostream
-template <typename T>
+  template <typename T,
+    typename std::enable_if<
+      !std::is_base_of<Expression, typename std::remove_pointer<T>::type>::value
```

Comment:
I do like that idea, but in this case I think it wouldn't change much; it would just remove the need for the `remove_ptr` bit.